server/service/src/programs/patient/patient_updated.rs

```diff
@@ -128,15 +149,79 @@ pub fn update_patient_row(
+        .unwrap_or(ContactDetails::default());
+    let draft_contact_details = ContactDetails {
+        address_1: patient.address1.clone(),
+        address_2: patient.address2.clone(),
+        country: patient.country.clone(),
+        email: patient.email.clone(),
+        phone: patient.phone.clone(),
+        website: patient.website.clone(),
+        ..contact_details.clone()
     };
+    SchemaPatient {
+        id: patient.id.clone(),
+        code: if patient.code == "" {
+            document_data.code
+        } else {
+            Some(patient.code.clone())
+        },
+        code_2: patient.national_health_number.clone(),
+        first_name: patient.first_name.clone(),
+        last_name: patient.last_name.clone(),
+        gender: patient.gender.as_ref().map(|gender| match gender {
+            Gender::Female => SchemaGender::Female,
+            Gender::Male => SchemaGender::Male,
+            Gender::Transgender => SchemaGender::Transgender,
+            Gender::TransgenderMale => SchemaGender::TransgenderMale,
+            Gender::TransgenderMaleHormone => SchemaGender::TransgenderMaleHormone,
+            Gender::TransgenderMaleSurgical => SchemaGender::TransgenderMaleSurgical,
+            Gender::TransgenderFemale => SchemaGender::TransgenderFemale,
+            Gender::TransgenderFemaleHormone => SchemaGender::TransgenderFemaleHormone,
+            Gender::TransgenderFemaleSurgical => SchemaGender::TransgenderFemaleSurgical,
+            Gender::Unknown => SchemaGender::Unknown,
+            Gender::NonBinary => SchemaGender::NonBinary,
+        }),
+        contact_details: if contact_details == draft_contact_details {
+            document_data.contact_details
+        } else {
+            let mut contacts = vec![draft_contact_details];
+            if let Some(contact_details) = document_data.contact_details {
+                contacts.extend(contact_details.into_iter().skip(1));
+            }
+            Some(contacts)
+        },
+        date_of_birth: patient
+            .date_of_birth
+            .map(|date| date.format("%Y-%m-%d").to_string()),
+        date_of_death: patient
+            .date_of_death
+            .map(|date| date.format("%Y-%m-%d").to_string()),
 
-    if is_sync_update {
-        name_repo.sync_upsert_one(&name_upsert)?;
-    } else {
-        name_repo.upsert_one(&name_upsert)?;
+        middle_name: document_data.middle_name,
+        date_of_birth_is_estimated: document_data.date_of_birth_is_estimated,
+        is_deceased: Some(patient.is_deceased || document_data.is_deceased.unwrap_or(false)),
+        notes: document_data.notes,
+        passport_number: document_data.passport_number,
+        socio_economics: document_data.socio_economics,
+        allergies: document_data.allergies,
+        birth_place: document_data.birth_place,
+        marital_status: document_data.marital_status,
+        contacts: document_data.contacts,
+        extension: document_data.extension,
```

Comment:
There are currently:
1) name row
2) common patient document data, which contains extra data not in the name row
3) patient document extension for country specific data not in the common patient document data

For the patient, the enrolment and the encounter data only the common fields that needs to be queried are in a table (but they all have additional common fields).

Think currently it wouldn't cause much problem moving the new common data fields from 2) to 3) so that the main patient document data can be mapped 1 to 1 to the name row. If needed we can then extent name_row and common patient document data type.

Main problem is that its used in the schemas and reports which would need to be updated and tested.

(Another thing to consider but which hasn't really be done is to have a FHIR friendly patient data structure.)